buildchain/buildchain/salt_tree.py

```diff
@@ -81,158 +81,147 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
-
-    Path('salt/metalk8s/calico/configured.sls'),
-    Path('salt/metalk8s/calico/deployed.sls'),
-    Path('salt/metalk8s/calico/init.sls'),
-    Path('salt/metalk8s/calico/installed.sls'),
-
-    Path('salt/metalk8s/containerd/configured.sls'),
-    Path('salt/metalk8s/containerd/init.sls'),
-    Path('salt/metalk8s/containerd/installed.sls'),
+    Path('salt/metalk8s/container-engine/containerd/configured.sls'),
+    Path('salt/metalk8s/container-engine/containerd/init.sls'),
+    Path('salt/metalk8s/container-engine/containerd/installed.sls'),
 
     Path('salt/metalk8s/defaults.yaml'),
 
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver-etcd-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver-kubelet-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-deploy-ca-cert.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-healthcheck-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-peer.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-server.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-deploy-ca-cert.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/installed.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/sa.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/sa-deploy-pub-key.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/control-plane/apiserver.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/controller-manager.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/files/htpasswd'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/files/manifest.yaml'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/scheduler.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/etcd/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/etcd/files/manifest.yaml'),
-    Path('salt/metalk8s/kubeadm/init/etcd/local.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/init.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/configured.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/files/kubeadm.env'),
+    Path('salt/metalk8s/_internal/certs/init.sls'),
+    Path('salt/metalk8s/_internal/certs/installed.sls'),
+    Path('salt/metalk8s/_internal/certs/removed.sls'),
+
+    Path('salt/metalk8s/_internal/files/control-plane-manifest.yaml'),
+
+    Path('salt/metalk8s/_internal/init.sls'),
+
+    Path('salt/metalk8s/_internal/preflight/init.sls'),
+    Path('salt/metalk8s/_internal/preflight/mandatory.sls'),
+    Path('salt/metalk8s/_internal/preflight/recommended.sls'),
+
+    Path('salt/metalk8s/_internal/python-kubernetes/init.sls'),
```

Comment:
Is this still required? There should be no need for `python-kubernetes` on nodes.

Reply:
Yes it's not executed on the nodes it's just part of the `deployed.sls` states

Reply:
The `salt-master` container has `python-kubernetes` already installed, as it should, so I guess this can now be removed entirely?

Reply:
Yes we can just for me it's a requirements of the state but I would say that kind of requirements it's the goal of virtual to handle this.
So yes i will